### PR TITLE
(Chore)[#114771085] Test Channel routes, controllers, model and repositories.

### PR DESCRIPTION
**Pull request by johnkariuki**
Add tests to check that all, deleted and active pages work.
Assert that all users can view active channels

### CODE REVIEW

**Comment by andela-oadebayo** [comment]
-Please work on writing comments in your test, i believe they are very much needed in Test, especially when you edit a previous test
-Work on test that need admin access, and make sure the admin access is being granted and not regular users are being used
-Nice Work well done

File: tests/ChannelTest.php [hunk 1]
```diff
@@ -13,14 +15,172 @@ class ChannelTest extends TestCase
...
+        $this->actingAs($user)
+         ->visit('/dashboard/channel/create')
+         ->type('Swanky new name', 'name')
+         ->type('Swanky new description', 'description')
+         ->press('create')
+         ->seeInDatabase('channels', [
+            'channel_name' => 'Swanky new name'
+         ]);
+
+         $user = factory(User::class)->create();
```
Comment (unicodeveloper) [inline tests/ChannelTest.php]: **Code Functionality - Critical**

Only  a user with the role of an admin or premium user should have the ability to create channels

File: tests/ChannelTest.php [hunk 2]
```diff
@@ -13,14 +15,172 @@ class ChannelTest extends TestCase
...
+    /**
+     * Assert that edit route updates the channel.
+     *
+     * @return void
+     */
+    public function testEditChannel()
+    {
+        $this->withoutMiddleware();
+
+        $user = factory(User::class)->create();
```
Comment (unicodeveloper) [inline tests/ChannelTest.php]: **Code Functionality - Critical**

Only a user with the role of an admin or premium user should have the ability to edit channels

**Comment by johnkariuki** [comment]
@unicodeveloper The highlighted changes have been made and should be good for merging now.

**Pull request merged**